analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/PublicMethodArgumentsShouldBeCheckedForNull.ParameterReassignedConstraintTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks.CSharp;
+using SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class ParameterReassignedConstraintTest
+{
+    [DataTestMethod]
+    [DataRow(@"arg = Unknown<object>();", "object")]
+    [DataRow(@"arg = Unknown<List<object>>();", "List<object>")]
+    [DataRow(@"arg = Unknown<int>();", "int")]
+    [DataRow(@"arg = Unknown<int?>();", "int?")]
+    [DataRow(@"arg = Unknown<object>();", "ref object")]
+    [DataRow(@"arg = Unknown<object>();", "out object")]
+    [DataRow(@"arg = Unknown<object[]>();", "params object[]")]
```

Comment:
Do you need the  type arguments? I don't think so. Type interference should do the job for you:
```cs
[DataRow(@"arg = Unknown();", "object")]
```
If this works, that part never changes and does not need to be parameterized.

Reply:
It doesn't work. Even on .NET 7.0 it still can't infer it.